src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/editdocs/EditDocsAuditMetadataTypeService.java

```diff
@@ -0,0 +1,44 @@
+package uk.gov.hmcts.reform.iacaseapi.domain.handlers.postsubmit.editdocs;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCase;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCaseFieldDefinition;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.DocumentWithMetadata;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.field.IdValue;
+
+@Component
+@Qualifier("documentWithMetadataAudit")
+public class EditDocsAuditMetadataTypeService implements EditDocsAuditService {
+    @Override
+    public List<String> getUpdatedAndDeletedDocIdsForGivenField(AsylumCase asylumCase, AsylumCase asylumCaseBefore,
+                                                                AsylumCaseFieldDefinition field) {
+        List<IdValue<DocumentWithMetadata>> doc = getDocField(asylumCase, field);
+        List<IdValue<DocumentWithMetadata>> docBefore = getDocField(asylumCaseBefore, field);
+        docBefore.removeAll(doc);
+        List<String> docIds = new ArrayList<>();
+        docBefore.forEach(d -> docIds.add(getIdFromDocUrl(d.getValue().getDocument().getDocumentUrl())));
+        return docIds;
+    }
+
+    public static String getIdFromDocUrl(String documentUrl) {
+        String regexToGetStringFromTheLastForwardSlash = "([^/]+$)";
```

Comment:
Not needed in this instance, but escaping the forward slash is a good practice `([^\/]+$)`. Makes the regex clearer in my opinion and in other languages patterns start with a `/` so might be ambiguous for some people reading the regex.

Reply:
that's good, did not know it, tbh. not my main strength. Thanks. TODO.

Reply:
![Screenshot 2020-03-17 at 22 26 57](https://user-images.githubusercontent.com/3930991/76903544-82fd9600-689e-11ea-9f45-ba6c186f699f.jpg)

as per the image above, I will leave it as it is on this occasion

Reply:
done